src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VepAnnotationFileWriter.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2015-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.io.writers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.ItemStreamWriter;
+
+import uk.ac.ebi.eva.pipeline.model.VariantWrapper;
+import uk.ac.ebi.eva.pipeline.parameters.AnnotationParameters;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.zip.GZIPOutputStream;
+
+public class VepAnnotationFileWriter implements ItemStreamWriter<VariantWrapper> {
+    private static final Logger logger = LoggerFactory.getLogger(VepAnnotationFileWriter.class);
+
+    private AnnotationParameters annotationParameters;
+
+    private Integer chunkSize;
+
+    private Process process;
+
+    private OutputStream perlStdin;
+
+    private final Long timeoutInSeconds;
+
+    private Thread outputCapturer;
+
+    private AtomicBoolean writingOk;
+
+    private boolean alreadyClosed;
+
+    public VepAnnotationFileWriter(Long timeoutInSeconds, Integer chunkSize, AnnotationParameters annotationParameters) {
```

Comment:
I agrre with Diego, It will be clearer if the concrete logic of VEP would be encapsulated in it's own class, or that when constructed returns a class that we use as interface or any similar solution.